egs/wsj/s5/steps/nnet3/lstm/train.sh

```diff
@@ -449,9 +459,21 @@ while [ $x -lt $num_iters ]; do
   this_num_jobs=$(perl -e "print int(0.5+$num_jobs_initial+($num_jobs_final-$num_jobs_initial)*$x/$num_iters);")
 
   ilr=$initial_effective_lrate; flr=$final_effective_lrate; np=$num_archives_processed; nt=$num_archives_to_process;
-  this_learning_rate=$(perl -e "print (($x + 1 >= $num_iters ? $flr : $ilr*exp($np*log($flr/$ilr)/$nt))*$this_num_jobs);");
+  this_effective_learning_rate=$(perl -e "print ($x + 1 >= $num_iters ? $flr : $ilr*exp($np*log($flr/$ilr)/$nt));");
+  this_learning_rate=$(perl -e "print ($this_effective_learning_rate*$this_num_jobs);");
+  if [ -z "$adaptive_shrink" ]; then
+    this_shrink=$shrink
+  else
+    if [ "$adaptive_shrink" == "type1" ]; then
+      this_shrink=$(perl -e "print 1.0 - (($this_learning_rate / $ilr) * (1 - $shrink));")
```

Comment:
(rewriting the comment as it was hidden due to an updated pull request)

this_shrink=$(perl -e "print 1.0 - (($this_learning_rate / $ilr) * (1 - $shrink));")
which depends on the actual learning rate leads to zigzag and highly varying shrink values even for $shrink=0.95 and 0.98. Both the below training runs diverge

red corresponds to $shrink=0.98 and blue 0.95
Plot1 shows shrink values
and plot2 cv accuracies.

![shrink_vs_cvaccuracy_iteration](https://cloud.githubusercontent.com/assets/3895456/10120126/aa0618d0-647a-11e5-8835-eac94b5db8db.png)